src/components/cylc/workflow/Toolbar.vue

```diff
@@ -357,9 +372,20 @@ export default {
         )
       }
     },
-    nEdgeDistance () {
-      // the graph window distance reported by the scheduler
-      return this.currentWorkflow?.node?.nEdgeDistance
+    nWindow: {
+      get () {
+        // the graph window distance reported by the scheduler
+        return this.currentWorkflow?.node?.nEdgeDistance ?? 1
+      },
+      async set (val) {
+        if (val == null || this.isStopped) return
+
+        this.changingNWindow = true
+        if (await this.setGraphWindow(val)) {
+          await until(() => this.currentWorkflow?.node?.nEdgeDistance === val)
+        }
+        this.changingNWindow = false
+      },
```

Comment:
By switching to a writeable computed property, the state relies on the subscription value and so cannot deviate from it, fixing the bug. But this means there is a delay in the update of the value displayed in the input after choosing a new value, so I added a loading spinner to give immediate feedback to the user